### PR TITLE
Account creation

### PR DESCRIPTION
**Pull request by DisparagingEgo1**
User Account Features Implementation

### CODE REVIEW

File: back-end/services/user.py [hunk 1]
```diff
@@ -82,21 +102,21 @@ def login(username, password):
...
+                response = {"message": "Logged in"}, 200
+                cnx.close()   
+                return response, 200   
         except Error as e:
             print("Error while connecting to MySQL", e)
             #log if any errors happen with connection
-            response = {"message": "Error while connecting to MySQL"}
+            response = {"message": "Error while connecting to MySQL"}, 500
             cnx.close()
             return response, 500
```
Comment (schuywalker) [inline back-end/services/user.py:117]:
```suggestion
            return response
```

Comment (schuywalker) [reply back-end/services/user.py:117]: response already has response code, so can we get rid of it here. I think a lot of end points have multiple response codes in them so may be a refactor thing.

File: back-end/services/user.py [hunk 2]
```diff
@@ -160,4 +180,60 @@ def deleteUser(user):
...
+            cursor.execute(query, (username,hashpass))
+            cursor.fetchall()
+            if(cursor.rowcount == 1):
+                cnx.close()
+                response={"message":"Correct Password"}, 200
+                return response
+            else:
+                cnx.close()
+                response = {"message": "Incorrect Password"}
+                return response, 400
```
Comment (schuywalker) [inline back-end/services/user.py:217]:
```suggestion
                return response, 401
```

File: back-end/endpoints/endpoints.py [hunk 3]
```diff
@@ -68,13 +68,17 @@ def get(self):
...
         data = UserService.login(request.args.get('username'), request.args.get('password'))
-        return data, 200         
+        return data, 200        
 
 class CreateUser(Resource):
     def post(self):
-        data = UserService.createUser(request.args.get('username'), request.args.get('password'), request.args.get('email'), request.args.get('first'), request.args.get('last'))
-        return data, 200
+        if(UserService.isUserNameAvailable(request.args.get('username'))):
+            data = UserService.createUser(request.args.get('username'), request.args.get('password'), request.args.get('email'), request.args.get('first'), request.args.get('last'))
```
Comment (schuywalker) [inline back-end/endpoints/endpoints.py:78]: technically i feel like this should be in a service but nbd/ probably not worth moving

File: back-end/services/watchlist.py [hunk 4]
```diff
@@ -90,99 +94,118 @@ def deleteWatchlist(wl_ID):
...
+            finally:
+                WatchlistService.updateWLTime(wl_ID, cnx, cursor)
+                cursor.close()
+                dbc.close()
         except Error as e:
-            if (e.errno == errorcode.ER_DUP_ENTRY):
-                return ("Error: Duplicate Entry",409)
-            else:
-                return ("Error: ",e,400)
+            response = {"message": "Error: "}, 500
```
Comment (schuywalker) [inline back-end/services/watchlist.py:172]:
```suggestion
            response = {"message": ("Error: "+e)}, 500
```

Comment (schuywalker) [reply back-end/services/watchlist.py:172]: with can also do e.printStackTrace()

**Review by schuywalker — APPROVED** [review]
good pull

**Comment by mszym99** [comment]
Looks good

**Review by mlorhan — APPROVED** [review]

**Pull request merged**